sw/device/lib/base/libgcc_polyfill_unittest.cc

```diff
@@ -0,0 +1,117 @@
+  {458834222u, 776034587u}, {1725702518u, 1981930598u}, {2658362673u, 828732318u}, {694561013u, 4113065513u},
+  {2234693117u, 4256510597u}, {2146340768u, 2693787263u}, {3036393057u, 1638857652u}, {515412848u, 1888729118u},
+  {1484651919u, 2415754584u}, {851907104u, 538101554u}, {1505051753u, 1766110553u}, {3408912721u, 1373712331u},
+  {1695131141u, 94767461u}, {3901279206u, 3872098536u}, {1644878809u, 3655010914u}, {3368358259u, 1930544584u},
+  {1094569037u, 1305230657u}, {215149339u, 468439564u}, {3955150114u, 583909099u}, {2147398144u, 11730559u},
+  {2508623926u, 915441301u}, {2924276618u, 2330676398u}, {3742087918u, 4005170143u}, {2271260870u, 3333447815u},
+};
+// clang-format on
+
+INSTANTIATE_TEST_SUITE_P(Bswap32Test, Bswap32Test,
+                         testing::ValuesIn(kBswap32Vectors));
+
+struct Popcount32Vector {
+  uint32_t a, b;
+  friend void operator<<(std::ostream &out, const Popcount32Vector &v) {
+    out << "{" << v.a << ", " << v.b << "}";
+  }
+};
+
+class Popcount32Test : public testing::TestWithParam<Popcount32Vector> {};
+
+TEST_P(Popcount32Test, Popcount32) {
+  EXPECT_EQ(libgcc_popcount_i32(GetParam().a), GetParam().b);
+}
+
+// clang-format off
+// Simple python snippet for generating vectors:
+//
+// import random
+// def popcnt(x):
+//   return bin(x).count('1')
+// for _ in range(0, 16):
+//   a1 = random.getrandbits(32)
+//   a2 = random.getrandbits(32)
+//   a3 = random.getrandbits(32)
+//   a4 = random.getrandbits(32)
+//   print(f"{{{a1}u, {popcnt(a1)}u}}, {{{a2}u, {popcnt(a2)}u}}, {{{a3}u, {popcnt(a3)}u}}, {{{a4}u, {popcnt(a4)}u}},")
+constexpr Popcount32Vector kPopcount32Vectors[] = {
+  {835382398u, 18u}, {989772317u, 20u}, {3077882061u, 17u}, {1629695575u, 15u},
+  {2455869383u, 16u}, {2170626028u, 15u}, {4139221225u, 18u}, {3776125387u, 15u},
+  {3004111611u, 19u}, {2963131007u, 19u}, {4045235776u, 14u}, {1643400886u, 17u},
+  {4150738286u, 22u}, {1400852238u, 19u}, {2418702364u, 12u}, {467992256u, 17u},
+  {3682650303u, 17u}, {378631108u, 14u}, {1013652703u, 19u}, {346940308u, 15u},
+  {2641016996u, 16u}, {1418420962u, 14u}, {1157001308u, 15u}, {2033326519u, 17u},
+  {158659135u, 17u}, {3752321663u, 23u}, {4115183622u, 14u}, {3442445400u, 16u},
+  {2548411999u, 21u}, {805054824u, 17u}, {3642282719u, 18u}, {831908166u, 15u},
+  {488175709u, 16u}, {822777404u, 13u}, {2833109194u, 16u}, {1220029187u, 12u},
+  {1309775921u, 12u}, {3274499747u, 16u}, {1810710453u, 22u}, {1556697528u, 15u},
+  {4036618587u, 18u}, {250843971u, 16u}, {1745670579u, 14u}, {848197615u, 20u},
+  {2130508332u, 21u}, {2080369275u, 25u}, {3401596301u, 15u}, {3973930182u, 19u},
+  {2780804239u, 20u}, {3698231936u, 14u}, {2509067735u, 18u}, {4066024874u, 18u},
+  {3915944937u, 17u}, {589196589u, 16u}, {3015647820u, 17u}, {3158734726u, 17u},
+  {4066116624u, 11u}, {1085462442u, 15u}, {2411105122u, 17u}, {844606662u, 16u},
+  {3200259285u, 15u}, {3503782720u, 16u}, {278363999u, 19u}, {532777955u, 17u},
```

Comment:
nit: if the first arg was expressed in binary/hex it would be clearer that/why it was correct.